src/infinite-scroll.js

```diff
@@ -80,21 +74,25 @@ angular.module(MODULE_NAME, [])
         }
 
         const remaining = elementBottom - containerBottom;
-        const shouldScroll = remaining <= (height(container) * scrollDistance) + 1;
-
-        if (shouldScroll) {
-          checkWhenEnabled = true;
+        return remaining <= (height(container) * scrollDistance) + 1;
+      }
 
-          if (scrollEnabled) {
-            if (scope.$$phase || $rootScope.$$phase) {
-              scope.infiniteScroll();
-            } else {
-              scope.$apply(scope.infiniteScroll);
-            }
+      // infinite-scroll specifies a function to call when the window,
+      // or some other container specified by infinite-scroll-container,
+      // is scrolled within a certain range from the bottom of the
+      // document. It is recommended to use infinite-scroll-disabled
+      // with a boolean that is set to true when the function is
+      // called in order to throttle the function call.
+      function defaultHandler() {
+        immediateCheck = false;
+        checkWhenEnabled = shouldScroll();
+        if (checkWhenEnabled && scrollEnabled) {
+          if (scope.$$phase || $rootScope.$$phase) {
+            scope.infiniteScroll();
+          } else {
+            scope.$apply(scope.infiniteScroll);
           }
-        } else {
-          if (checkInterval) { $interval.cancel(checkInterval); }
-          checkWhenEnabled = false;
+          $timeout(handler);
```

Comment:
not fully recursive as we have timeout and throttle in between but the idea is to check again after the digest is done.
Digest should make the dom change son that digest or trigger scrollEnabled = false if it is async calls.